convert/convert_test.go

```diff
@@ -0,0 +1,161 @@
+package convert
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func TestConvertBsonBytes(t *testing.T) {
+	doc := bson.M{
+		"op": "i",
+		"ns": "throttle.test",
+		"o": bson.M{
+			"_id": bson.NewObjectId(),
+			"val": "55d57fd49e8a1b0d007f73b4",
+		},
+	}
+	bytes, err := bson.Marshal(doc)
+	assert.NoError(t, err)
+
+	op, err := OplogBytesToOp(bytes)
+	assert.NoError(t, err)
+	assert.Equal(t, "insert", op.Type)
+}
+
+func TestConvertInsertOp(t *testing.T) {
+	obj := bson.M{
+		"_id": "teacherId",
+		"val": "55d57fd49e8a1b0d007f73b4",
+	}
+	doc := bson.M{
+		"v":  2,
```

Comment:
i think i'm going to hold off on 3.0 support for now. hopefully it won't be too hard, but i don't have a 3.0 oplog easily accessible